pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/operator/V2Operator.java

```diff
@@ -0,0 +1,49 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.query.runtime.operator;
+
+import java.util.List;
+import org.apache.pinot.core.common.Operator;
+import org.apache.pinot.core.operator.BaseOperator;
+import org.apache.pinot.query.runtime.blocks.TransferableBlock;
+import org.slf4j.LoggerFactory;
+
+
+public abstract class V2Operator extends BaseOperator<TransferableBlock> implements AutoCloseable {
+  private static final org.slf4j.Logger LOGGER = LoggerFactory.getLogger(MailboxReceiveOperator.class);
+
+  @Override
+  public List<Operator> getChildOperators() {
```

Comment:
instead we can change the super class implementation to be `public List<? extends Operator> getChildOperators()` and then I think we can have this class just specify that it needs to be `List<V2Operator>`

Reply:
Using  public List<? extends Operator> getChildOperators()  means we have to do type conversion and type check for every call to call ops in V2 interface. So I think it is easier to just have a separate API. WDTY?

Reply:
oh. I see what you meant. You were saying changing the return type to List<? extends Operator> for the API. Makes sense. Let me update the API.

Reply:
+1

Reply:
Sorry, I don't know Java very well. It looks like if we change the API, we need to update all of the callers to this method too? I added a TODO. I am happy to add it as a follow up. I don't want to touch all of the files in this PR.

Reply:
i would rather have this issue addressed since we are introducing a new abstract methods that will soon be deprecated. 

here is my findings. 
1. @agavra 's suggestion is to add a capture group for the return on the Operator API. 
```
public List<? extends Operator> getChildOperators();
```
the reason why is b/c Override cannot be done on direct `List<Operator>` since there's a type-erasure. CC @siddharthteotia for more info.

2. the issue of the API change occurs when `PlanNode.run()` where it returns a `Operator<? extends Block>` instead of a type-erasure `Operator`. this will cause trouble. in order to solve this. any usage of 
```
queryPlan.getPlanNode().run().getChildOperators()
```
should be split into:
```
Operator operator = queryPlan.getPlanNode().run(); // here we explicitly remove the generic.
List<Operator> childOperators = operator.getChildOperators();
```

and upon checking there's only one usage of this in ServerQueryExecutorV1Impl so it should be an easy fix. let me know if you have any questions @61yao

Reply:
Done